### PR TITLE
Make IsPsiOperative() function accept a list of defined Soldier classes

### PR DESCRIPTION
**Pull request by boundir**
This will let defined classes use Psi PCS #602

### CODE REVIEW

**Comment by robojumper** [comment]
Are there any places in the base game where we want to replace `UnitState.GetSoldierClassTemplateName() == 'PsiOperative'` with `UnitState.IsPsiOperative()` now that they aren't equivalent?

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/CHHelpers.uc [hunk 1]
```diff
@@ -434,3 +438,19 @@ static function array<name> GetAmbushRiskTemplateNames()
...
+	local Name ClassName;
+
+	foreach default.CLASS_AS_PSI_CLASS(ClassName)
+	{
+		if(SoldierClass == ClassName)
+		{
+			return true;
+		}
+	}
+	return false;
```
Comment (MrNiceUK) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/CHHelpers.uc]: There's a much shorter way to check an array for a particular value:
```
return default.CLASS_AS_PSI_CLASS.Find(ClassName) != INDEX_NONE;
```

**Comment by MrNiceUK** [comment]
Quick check for `'PsiOperative'` in the code and where it's used to check the soldier class, some MP/Skirmish checks which probably don't matter. In the single player campaign, seem all related to the psi-training mechanic and related issues (for completeness, also the photobooth has a select/case block which goes through the base game classes too).

**Comment by pledbrook** [comment]
There are quite a few significant locations that check for the soldier class directly:

 * `UIFacility_PsiLabSlot.OnPersonnelSelected()`
* `X2DynamicNarrative_DefaultDynamicNarratives.CreateTemplates()` (perhaps not important)
* `X2StrategyElement_DefaultStaffSlots.ShouldDisplayPsiChamberSoldierToDoWarning()`
* `X2StrategyElement_DefaultStaffSlots.IsUnitValidForPsiChamberSoldierSlot()` (but can be overridden by event)
* `X2StrategyElement_DefaultStaffSlots.GetPsiChamberSoldierBonusDisplayString()`
* `XComGameState_HeadquartersProjectPsiTraining.SetProjectFocus()`
* `XComGameState_HeadquartersXCom.MeetsSoldierRankGates()`
* `XComGameState_HeadquartersXCom.GetPsiTrainingEvents()`
* `XComGameState_MissionSite.ResumePsiOperativeTraining()`
* `XComGameState_Unit.HasAvailablePerksToAssign()`
* `XComGameState_Unit.RankUpSoldier()`
* `XComGameStateContext_HeadquartersOrder.CompleteUnitHealing()`
* `XComGameStateContext_HeadquartersOrder.CompletePsiTraining()`
* `XComGameStateContext_StrategyGameRule.PostMissionUpdatePsiOperativeTraining()`
* `XComHQPresentationLayer.ShowPromotionUI()`
* `XComUnitPawnNativeBase.PassesPlayLimitsTestUnit()`

Got to love the consistency.

**Comment by pledbrook** [comment]
So really, this is a question of what the actual intent of the change is. If it's simply to allow soldier classes to use the Psi PCS, then I would go with what I suggested in #604: add a configurable array of soldier classes that can explicitly use that PCS. The proposed change in this pull request is much more fundamental and requires a review of usages of `IsPsiOperative()` and the check for `'PsiOperative'` soldier class.

**Comment by boundir** [comment]
I don't know why we wouldn't want the change to affect `UISoldierHeader.uc` where it seems to be used to display stats.
I made this change because it didn't seem to affect facility training or functions specifically related to 'PsiOperative' class. 

Maybe what we want then is to replace calls to `IsPsiOperative()` where we want to allow access to other classes. That way we could safely replace `UnitState.GetSoldierClassTemplateName() == 'PsiOperative'` with `UnitState.IsPsiOperative()` as robo suggested.

**Comment by Xymanek** [comment]
I think this PR has too many unresolved comments, I do not consider it ready to be part of the stable CHL yet.

Suggestion: revert

**Comment by robojumper** [comment]
This change was **reverted** on master.

**Pull request merged**